src/core/main.c

```diff
@@ -2123,18 +2137,31 @@ static int parse_configuration(void) {
                         log_warning_errno(r, "Failed to parse kernel command line, ignoring: %m");
         }
 
+        /* Initialize some default rlimits for services */
+        if (!arg_default_rlimit[RLIMIT_NOFILE])
```

Comment:
please move this test into the function itself, so that it just exits early when there's already something set. i.e. hide the complexity in the callee, not the caller, where possible

Reply:
done